sklearn/calibration.py

```diff
@@ -209,31 +271,31 @@ def fit(self, X, y, sample_weight=None):
         else:
             cv = check_cv(self.cv, y, classifier=True)
             fit_parameters = signature(base_estimator.fit).parameters
-            base_estimator_supports_sw = "sample_weight" in fit_parameters
+            supports_sw = "sample_weight" in fit_parameters
 
             if sample_weight is not None:
                 sample_weight = _check_sample_weight(sample_weight, X)
 
-                if not base_estimator_supports_sw:
+                if not supports_sw:
                     estimator_name = type(base_estimator).__name__
                     warnings.warn("Since %s does not support sample_weights, "
                                   "sample weights will only be used for the "
                                   "calibration itself." % estimator_name)
 
-            for train, test in cv.split(X, y):
-                this_estimator = clone(base_estimator)
-
-                if sample_weight is not None and base_estimator_supports_sw:
-                    this_estimator.fit(X[train], y[train],
-                                       sample_weight=sample_weight[train])
-                else:
-                    this_estimator.fit(X[train], y[train])
-
-                calibrated_classifier = _CalibratedClassifier(
-                    this_estimator, method=self.method, classes=self.classes_)
-                sw = None if sample_weight is None else sample_weight[test]
-                calibrated_classifier.fit(X[test], y[test], sample_weight=sw)
-                self.calibrated_classifiers_.append(calibrated_classifier)
+            parallel = Parallel(n_jobs=self.n_jobs, verbose=self.verbose,
+                                pre_dispatch=self.pre_dispatch)
+
+            with parallel:
```

Comment:
I do not think we need this context manager, since there is only a single call to `parallel` here. The context manager is used when we want to [reuse a pool of workers](https://joblib.readthedocs.io/en/latest/parallel.html#reusing-a-pool-of-workers)